pkg/ovsdb/monitors.go

```diff
@@ -207,63 +271,78 @@ func (u *updater) prepareTableUpdate(event *clientv3.Event) (table string, uuid
 		if len(data) > 0 {
 			if event.IsCreate() {
 				if !u.isV1 {
-					rowUpdate = &ovsjson.RowUpdate{Insert: &data}
-				} else {
-					rowUpdate = &ovsjson.RowUpdate{New: &data}
+					return &ovsjson.RowUpdate{Insert: &data}, nil
 				}
-			} else {
-				// the delete for !u.isV1 we have returned before
-				rowUpdate = &ovsjson.RowUpdate{Old: &data}
+				return &ovsjson.RowUpdate{New: &data}, nil
 			}
+			// the delete for !u.isV1 we have returned before
+			return &ovsjson.RowUpdate{Old: &data}, nil
 		}
-	} else { // the event is modify
-		value = event.Kv.Value
-		data := map[string]interface{}{}
-		if err = json.Unmarshal(value, &data); err != nil {
-			return
-		}
-		prevValue := event.PrevKv.Value
-		prevData := map[string]interface{}{}
-		if err = json.Unmarshal(prevValue, &prevData); err != nil {
-			return
-		}
-		delete(data, "uuid")
-		delete(prevData, "uuid")
-		for column, cValue := range data {
-			if len(u.Columns) != 0 {
-				if _, ok := u.Columns[column]; !ok {
-					delete(data, column)
-					delete(prevData, column)
-					continue
-				}
-			}
-			if reflect.DeepEqual(cValue, prevData[column]) {
-				// TODO compare sets and maps
-				if u.isV1 {
-					delete(prevData, column)
-				} else {
-					delete(data, column)
-				}
+		return nil, nil
+	}
+	// the event is modify
+	if !u.Select.Modify {
+		return nil, nil
+	}
+	data, err := unmarshalData(event.Kv.Value)
+	if err != nil {
+		return nil, err
+	}
+	prevData, err := unmarshalData(event.PrevKv.Value)
+	if err != nil {
+		return nil, err
+	}
+	for column, cValue := range data {
+		if len(u.Columns) != 0 {
+			if _, ok := u.Columns[column]; !ok {
+				delete(data, column)
+				delete(prevData, column)
+				continue
 			}
 		}
-		if !u.isV1 {
-			if len(data) > 0 {
-				rowUpdate = &ovsjson.RowUpdate{Modify: &data}
-			}
-		} else {
-			if len(prevData) > 0 { // there are monitored updates
-				rowUpdate = &ovsjson.RowUpdate{New: &data, Old: &prevData}
+		if reflect.DeepEqual(cValue, prevData[column]) {
+			// TODO compare sets and maps
+			if u.isV1 {
+				delete(prevData, column)
+			} else {
+				delete(data, column)
 			}
 		}
 	}
-	return
+	if !u.isV1 {
+		if len(data) > 0 {
+			return &ovsjson.RowUpdate{Modify: &data}, nil
+		}
+	} else {
+		if len(prevData) > 0 { // there are monitored updates
+			return &ovsjson.RowUpdate{New: &data, Old: &prevData}, nil
+		}
+	}
+	return nil, nil
 }
 
-func keyToTableAndUUID(key string) (table string, uuid string, err error) {
+func unmarshalData(data []byte) (map[string]interface{}, error) {
+	obj := map[string]interface{}{}
+	if err := json.Unmarshal(data, &obj); err != nil {
+		return nil, err
+	}
+	delete(obj, "uuid")
```

Comment:
should it not be `_uuid` ? if so then maybe define in a constant

Reply:
also there are other specifcal fields that maybe you want to delete such as `_version`

Reply:
Agree about uuid constant and about  _uuid. I've started to think, that maybe we don't need this (_uuid) column at all, we have is as part of the key.
`_version` is returned but OVSDB